File: vespalib/src/vespa/vespalib/util/host_name.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "host_name.h"
#include <vespa/defaults.h>

namespace vespalib {

namespace {

vespalib::string make_host_name() {
    return vespa::Defaults::vespaHostname();
}

} // namespace vespalib::<unnamed>

const vespalib::string HostName::_host_name = make_host_name();

} // namespace vespalib

File: vespajlib/src/main/java/ai/vespa/validation/Hostname.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package ai.vespa.validation;

import java.util.regex.Pattern;

import static ai.vespa.validation.Validation.requireMatch;

/**
 * A valid hostname.
 *
 * @author jonmv
 */
public class Hostname extends StringWrapper<Hostname> {

    public static final Pattern hostnameLabel = Pattern.compile("([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]{0,61}[A-Za-z0-9])");
    public static final Pattern hostnamePattern = Pattern.compile("(?=.{1,255})(" + hostnameLabel + "\\.)*" + hostnameLabel);

    private Hostname(String value) {
        super(value);
    }

    public static Hostname of(String hostname) {
        return new Hostname(requireMatch(hostname, "hostname", hostnamePattern));
   }

    public static String requireLabel(String label) {
        return requireMatch(label, "hostname label", hostnameLabel);
    }

}
